tools/export/cces/__init__.py

```diff
@@ -0,0 +1,418 @@
+
+        proc = t.device_name
+        fpu = None
+        float_abi = None
+
+        if t.core == "Cortex-M4F":
+            cpu = "cortex-m4"
+            fpu = "fpv4-sp-d16"
+            float_abi = "softfp"
+        else:
+            cpu = t.core.lower()
+
+        asm_defines = self.toolchain.get_symbols(True)
+        c_defines = self.toolchain.get_symbols()
+
+        include_dirs = [self.format_path(d) for d in self.resources.inc_dirs if d]
+
+        srcs = self.resources.s_sources + \
+                self.resources.c_sources + \
+                self.resources.cpp_sources + \
+                self.resources.headers
+
+        srcs_dict = {}
+        for src in srcs:
+            srcs_dict[src] = self.format_path(src)
+
+        ld_script = self.format_path(self.resources.linker_script)
+
+        asm_flags = self.flags['asm_flags']
+        c_flags = self.flags['c_flags'] + self.flags['common_flags']
+        cxx_flags = self.flags['cxx_flags'] + self.flags['common_flags']
+
+        libs = []
+        for libpath in self.resources.libraries:
+            lib, ext = os.path.splitext(os.path.basename(libpath))
+            libs.append(lib[3:]) # skip 'lib' prefix
+
+        system_libs = [
+            '-lstdc++', '-lsupc++', '-lc', '-lgcc'
+            ]
+
+        ld_flags = self.flags['ld_flags'] + system_libs
+
+        flags = ["-mcpu="+cpu, "-mthumb"]
+        if fpu is not None:
+            flags.append("-mfpu="+fpu)
+            flags.append("-mfloat-abi="+float_abi)
+
+        self.clean_flags(c_flags, flags)
+        self.clean_flags(cxx_flags, flags)
+        self.clean_flags(ld_flags, flags)
+
+        ld_opts = self.convert_linker_options(ld_flags)
+        asm_opts = self.convert_assembler_options(asm_flags)
+        c_opts = self.convert_compiler_options(c_flags)
+        cxx_opts = self.convert_compiler_options(cxx_flags)
+
+        workspace = tempfile.mkdtemp()
+        project = "cces"
+        json = "cces.json"
+        local_location = self.format_path(project)
+
+        jinja_ctx = {
+            'project' : self.project_name,
+            'cpu' : cpu,
+            'proc' : proc,
+            'family' : "ARM",
+            'asm_defines' : asm_defines,
+            'c_defines' : c_defines,
+            'fpu' : fpu,
+            'float_abi' : float_abi,
+            'ld_script' : ld_script,
+            'local_location' : local_location,
+            'srcs': srcs_dict,
+            'include_dirs' : include_dirs,
+            'ld_opts' : ld_opts,
+            'ld_flags' : ld_flags,
+            'asm_opts' : asm_opts,
+            'asm_flags' : asm_flags,
+            'c_opts' : c_opts,
+            'c_flags' : c_flags,
+            'cxx_opts' : cxx_opts,
+            'cxx_flags' : cxx_flags,
+        }
+
+        self.gen_file('cces/cces.json.tmpl', jinja_ctx,
+                      json, trim_blocks=True, lstrip_blocks=True)
+
+        # import .json file using CCES headless builder
```

Comment:
You cannot run IDE commons in an exporter. Please remove the following section of code. It looks like this was intended for the `build` method, and eventually made its way there.

Reply:
Thanks for the review. I'm not sure I understand this comment but perhaps some clarification on our end will help. Here we generate the cces.json file as an input to our headlesstools application to generate the project files ('-command projectcreate'). There isn't any building done at this time - only project creation. The user can then import the project into the IDE for building / debugging, or use the build method (note the differences in command line arguments). Does this help clarify things, or am I misunderstanding your comment?

Reply:
> Does this help clarify things, or am I misunderstanding your comment?

Yes. This is still not allowed in exporters.

The online IDE does not have your IDE's executable installed. How can it be expected to handle this part of the export process?

Reply:
Understood - I hadn't considered the online IDE. Thanks for the clarification.